metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/feed/JcrFeedProvider.java

```diff
@@ -708,6 +728,66 @@ public void populateInverseFeedDependencies() {
                 .forEach(depFeed -> depFeed.addUsedByFeed(feed1));
         });
     }
+    
+    @Override
+    protected void appendJoins(StringBuilder bldr, String filter) {
+        if (! Strings.isNullOrEmpty(filter)) {
+            bldr.append("JOIN [tba:feedSummary] AS fs ON ISCHILDNODE(fs, e) ");
+            bldr.append("JOIN [tba:categoryDetails] AS cd ON ISCHILDNODE(e, cd) ");
+            bldr.append("JOIN [tba:category] AS c ON ISCHILDNODE(cd, c) ");
+            bldr.append("JOIN [tba:feedData] AS fd ON ISCHILDNODE(fd, e) ");
+//            bldr.append("JOIN [tba:feedTemplate] AS t ON t.[jcr:uuid] = fd.[tba:feedTemplate] ");
+        }
+    }
+    
+    @Override
+    protected void appendJoins(StringBuilder bldr, Pageable pageable, String filter) {
+        List<String> sortProps = new ArrayList<>();
+        pageable.getSort().forEach(o -> sortProps.add(o.getProperty()));
+        
+        if (! Strings.isNullOrEmpty(filter)) {
+            appendJoins(bldr, filter);
+        } else if (sortProps.contains(SORT_FEED_NAME)) {
+            bldr.append("JOIN [tba:feedSummary] AS fs ON ISCHILDNODE(fs, e) ");
+        } else if (sortProps.contains(SORT_CATEGORY_NAME)) {
+            bldr.append("JOIN [tba:categoryDetails] AS cd ON ISCHILDNODE(e, cd) ");
+            bldr.append("JOIN [tba:category] AS c ON ISCHILDNODE(cd, c) ");
+        } else if (sortProps.contains(SORT_TEMPLATE_NAME)) {
+            // NOTE: This join is not working - results in an empty result set
+            bldr.append("JOIN [tba:feedSummary] AS fs ON ISCHILDNODE(fs, e) ");
+            bldr.append("JOIN [tba:feedDetails] AS fd ON ISCHILDNODE(fd, fs) ");
+            bldr.append("JOIN [tba:feedTemplate] AS t ON t.[jcr:uuid] = fd.[tba:feedTemplate] ");
+        } else if (sortProps.contains(SORT_STATE)) {
+            bldr.append("JOIN [tba:feedData] AS fd ON ISCHILDNODE(fd, e) ");
+        }
+    }
+    
+    @Override
+    protected void appendFilter(StringBuilder bldr, String filter) {
+        String filterPattern = Strings.isNullOrEmpty(filter) ? null : "'%" + filter + "%'";
+        if (filterPattern != null) {
+            bldr.append("WHERE LOWER(").append(JCR_PROP_MAP.get(SORT_FEED_NAME)).append(") LIKE ").append(filterPattern);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal ") LIKE " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)